File: backend/schemas/procesos.py
"""Schemas para el Panel de Procesos (solo lectura)."""
from typing import List
from pydantic import BaseModel


class EstadoConteo(BaseModel):
    estado: str
    label: str
    total: int


class ProcesoResumen(BaseModel):
    proceso: str
    label: str
    tabla: str
    estados: List[EstadoConteo]
    total: int


class ProcesosResponse(BaseModel):
    procesos: List[ProcesoResumen]

File: frontend/services/procesos.ts
import { apiFetch } from "./api"

export interface EstadoConteo {
  estado: string
  label: string
  total: number
}

export interface ProcesoResumen {
  proceso: string
  label: string
  tabla: string
  estados: EstadoConteo[]
  total: number
}

export interface ProcesosData {
  procesos: ProcesoResumen[]
}

export function fetchProcesos(): Promise<ProcesosData> {
  return apiFetch<ProcesosData>("/api/procesos")
}
